src/EnergyPlus/WeatherManager.cc

```diff
@@ -2216,56 +2061,52 @@ namespace WeatherManager {
-                    state.dataWeatherManager->DDSkyTempScheduleValues(state.dataGlobal->TimeStep, state.dataGlobal->HourOfDay, envrnDayNum);
+
+            for (int iDD = 1; iDD <= state.dataEnvrn->TotDesDays; ++iDD) {
+                state.dataWeather->spSiteSchedules(iDD) = {-999.0, -999.0, -999.0, -999.0, -999.0};
+            }
+            
+            auto const &envCurr = state.dataWeather->Environment(state.dataWeather->Envrn);
+            int const envrnDayNum = envCurr.DesignDayNum;
+            auto &desDayInput = state.dataWeather->DesDayInput(envrnDayNum);
+            auto &spSiteSchedule = state.dataWeather->spSiteSchedules(envrnDayNum);
+            auto &desDayMod = state.dataWeather->desDayMods(envrnDayNum)(state.dataGlobal->TimeStep, state.dataGlobal->HourOfDay);
+            
+            if (desDayInput.dryBulbRangeType != DesDayDryBulbRangeType::Default) {
+                spSiteSchedule.OutDryBulbTemp = desDayMod.OutDryBulbTemp;
+            }
+
+            if (desDayInput.HumIndType == DesDayHumIndType::WBProfDef || desDayInput.HumIndType == DesDayHumIndType::WBProfDif ||
+                desDayInput.HumIndType == DesDayHumIndType::WBProfMul || desDayInput.HumIndType == DesDayHumIndType::RelHumSch) {
+                spSiteSchedule.OutRelHum = desDayMod.OutRelHum;
+            }
+            if (desDayInput.solarModel == DesDaySolarModel::SolarModel_Schedule) {
+                spSiteSchedule.BeamSolarRad = desDayMod.BeamSolarRad;
+                spSiteSchedule.DifSolarRad = desDayMod.DifSolarRad;
+            }
+
+            if (envCurr.skyTempModel == SkyTempModel::ScheduleValue || envCurr.skyTempModel == SkyTempModel::DryBulbDelta ||
+                envCurr.skyTempModel == SkyTempModel::DewPointDelta) {
+                spSiteSchedule.SkyTemp = desDayMod.SkyTemp;
             }
         } else if (state.dataEnvrn->TotDesDays > 0) {
-            state.dataWeatherManager->SPSiteDryBulbRangeModScheduleValue = -999.0;   // N/A Drybulb Temperature Range Modifier Schedule Value
-            state.dataWeatherManager->SPSiteHumidityConditionScheduleValue = -999.0; // N/A Humidity Condition Schedule Value
-            state.dataWeatherManager->SPSiteBeamSolarScheduleValue = -999.0;         // N/A Beam Solar Schedule Value
-            state.dataWeatherManager->SPSiteDiffuseSolarScheduleValue = -999.0;      // N/A Diffuse Solar Schedule Value
-            state.dataWeatherManager->SPSiteSkyTemperatureScheduleValue = -999.0;    // N/A SkyTemperature Modifier Schedule Value
+            for (int iDD = 1; iDD <= state.dataEnvrn->TotDesDays; ++iDD) {
+                state.dataWeather->spSiteSchedules(iDD) = {-999.0, -999.0, -999.0, -999.0, -999.0};
+            }
         }
 
-        state.dataEnvrn->WindSpeed = state.dataWeatherManager->TodayWindSpeed(state.dataGlobal->TimeStep, state.dataGlobal->HourOfDay);
+        state.dataEnvrn->WindSpeed = today.WindSpeed;
```

Comment:
These have been converted from individual arrays to arrays of structs where each struct represents all of the conditions for the time step.